packages/devtools_app/lib/src/framework/framework_core.dart

```diff
@@ -178,20 +168,24 @@ Future<void> _initDTDConnection() async {
             'Failed to connect to the Dart Tooling Daemon',
             isReportable: false,
           );
-          reportError(
+          error_handling.reportError(
             e,
             errorType: 'Dart Tooling Daemon connection failed.',
             stack: st,
           );
         },
       );
+
+      if (dtdManager.connection.value != null) {
+        ThemeManager(dtdManager.connection.value!).listenForThemeChanges();
```

Comment:
instead of calling a method on a new class to initialize the event handler, perhaps we should add an optional parameter `void Function(DtdEvent) onEvent` to `dtdManager.connect` where we can pass a callback to initialize an event handler.

This stream subscription will need to be stored on the DTDManager class so that it can be cancelled on `disconnect`.

Reply:
Does this mean `editorClient.event` needs to be a var on DTD manager and set to null on disconnect? I'm not sure how to structure this because `DtdEditorClient` is within `devtools_app` while `DTDManager` is in `devtools_app_shared`. Does this mean `DtdEditorClient` would need to be moved to the shared package as well?

Reply:
Ah okay I see what you are saying. The only thing that is unfortunate about having this as a separate class is that disposing the stream subscription has to take place separately instead of that being tied to the lifecycle of dtdManger. But this can be done separately.

To do this, mixin add the following to the ThemeManager class signature:
`extends DisposableController with AutoDisposeControllerMixin`

Then add the stream subscription with `addAutoDisposeStreamSubscription`. 

And then just make sure you call `ThemeManager.dispose` in the `FrameworkCore.dispose` method before disposing `dtdManager`.

Reply:
Thanks! Should `ThemeManager` be stored in `globals.dart` or can it be just in this file if it's not referenced elsewhere?

Reply:
Just in this file should be fine since it is not referenced elsewhere. Globals are generally used for Objects we want accessible from all of DevTools, and this object is just needed for the framework initialization.